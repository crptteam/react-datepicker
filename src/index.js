import DatePicker from './components/DatePicker/';
import RangePicker from './components/RangePicker/';

export { DatePicker, RangePicker };

export default DatePicker;