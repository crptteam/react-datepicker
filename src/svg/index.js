import CalendarIcon from './calendar-icon.svg';
import LeftDatepickerArrow from './left-datepicker-arrow.svg';
import RightDatepickerArrow from './right-datepicker-arrow.svg';

export { CalendarIcon, LeftDatepickerArrow, RightDatepickerArrow };
