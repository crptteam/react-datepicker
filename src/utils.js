import moment from 'moment';

moment.locale('ru');

export function getValidMomentFromISOStringOrNull(str) {
  const m = moment(str);
  return m.isValid() ? m : null;
}

export function getDaysArrayFromMomentDate(date) {
  const count = date.daysInMonth();
  const startOfMonth = date.startOf('month');
  const dayNumber = startOfMonth.isoWeekday();
  let currentDay;
  const days = [];

  for (let i = 2; i <= count + dayNumber; i++) {
    if (i < dayNumber) {
      days.push({ val: 0 });
    } else {
      currentDay = moment(startOfMonth).add(i - dayNumber - 1, 'day');

      days.push({
        val: i - dayNumber,
        date: currentDay,
        formated: currentDay.format('YYYY-MM-DD')
      });
    }
  }
  return days;
}

export function getAllPlainValuesAsObj(obj) {
  const plain = {};
  Object.keys(obj).forEach(key => typeof obj[key] !== "object" ? plain[key] = obj[key] : null);
  return plain;
}

export function getThemeAsPlainTextByKeys(theme, ...keys) {
  const plain = getAllPlainValuesAsObj(theme);

  keys.forEach(key => theme[key] && Object.assign(plain, theme[key]));

  return plain;
}


export function innerMerge(obj, ...others) {
  others.forEach(v => {
    for (const key in v) {
      if (typeof obj[key] === "object" && typeof v[key] === "object") {
        obj[key] = innerMerge({}, obj[key], v[key]);
      } else {
        obj[key] = v[key];
      }
    }
  });

  return obj;
}