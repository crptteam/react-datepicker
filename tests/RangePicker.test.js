import React from "react";
import { RangePicker } from "../src/";

describe("RangePicker", () => {

  it("Should renders without problems", () => {

    const wrapper = shallow(
      <RangePicker/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

