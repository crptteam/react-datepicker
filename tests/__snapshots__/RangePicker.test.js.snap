// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangePicker Should renders without problems 1`] = `
<RangePicker
  disabled={false}
  from={null}
  onChange={[Function]}
  onUpdate={[Function]}
  theme={
    Object {
      "DatePicker": Object {
        "Day": Object {
          "borderRadius": "2px",
          "height": "24px",
          "hoverBackground": "#1A99F4",
          "hoverColor": "#FFFFFF",
          "hovered": Object {
            "background": "rgb(95,191,255)",
            "color": "#FFFFFF",
          },
          "main": Object {
            "background": "rgba(0,0,0,0)",
            "color": "#212C42",
          },
          "selected": Object {
            "background": "#1A99F4",
            "color": "#FFFFFF",
          },
          "width": "24px",
        },
        "InputElem": Object {
          "disabled": Object {
            "background": "rgba(0,0,0,0)",
            "color": "#212C42",
            "cursor": "not-allowed",
            "placeholderColor": "#ACADB5",
          },
          "error": Object {
            "background": "rgba(0,0,0,0)",
            "color": "#212C42",
            "cursor": "text",
            "placeholderColor": "#ACADB5",
          },
          "main": Object {
            "background": "rgba(0,0,0,0)",
            "color": "#212C42",
            "cursor": "text",
            "placeholderColor": "#ACADB5",
          },
        },
        "InputWrap": Object {
          "borderRadius": "2px",
          "disabled": Object {
            "background": "#FFFFFF",
            "border": "1px solid #ABADB5",
            "cursor": "not-allowed",
          },
          "error": Object {
            "background": "#FFEBEB",
            "border": "1px solid #FF3C21",
            "cursor": "normal",
          },
          "height": "64px",
          "main": Object {
            "background": "#FFFFFF",
            "border": "1px solid #ABADB5",
            "cursor": "normal",
          },
          "paddingLeft": "16px",
          "paddingRight": "16px",
        },
        "borderRadius": "2px",
        "disabled": Object {
          "background": "#FFFFFF",
          "border": "1px solid #ABADB5",
          "color": "#212C42",
          "placeholder": "#ACADB5",
        },
        "error": Object {
          "background": "#FFEBEB",
          "border": "1px solid #FF3C21",
          "color": "#212C42",
          "placeholder": "#ACADB5",
        },
        "fontSize": "16px",
        "fontWeight": 200,
        "height": "64px",
        "main": Object {
          "background": "#FFFFFF",
          "border": "1px solid #ABADB5",
          "color": "#212C42",
          "placeholder": "#ACADB5",
        },
      },
    }
  }
  to={null}
/>
`;
